apis/secretmanager/v1beta1/secret_identity.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1beta1
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/GoogleCloudPlatform/k8s-config-connector/apis/common"
+	refsv1beta1 "github.com/GoogleCloudPlatform/k8s-config-connector/apis/refs/v1beta1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type SecretIdentity struct {
+	id     string
+	parent *SecretParent
+}
+
+func (i *SecretIdentity) String() string {
+	return i.parent.String() + "/secrets/" + i.id
+}
+
+func (r *SecretIdentity) Parent() *SecretParent {
+	return r.parent
+}
+
+func (r *SecretIdentity) ID() string {
+	return r.id
+}
+
+type SecretParent struct {
+	ProjectID string
```

Comment:
Are we planning to add to SecretParent? If not we could probably drop the struct as per https://go.dev/play/p/D8y1EF81zyF